design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +281,258 @@ permissionScope: Cluster
+
+We will model our roles to define a clear separation of concerns for cluster(environment) scoped resources,
+ and namespaced resources. In our model we assume that any Cluster Stack resources are global will be for the
+ environment or infrastructure scope, whereas any namespace stack resources will be for the namespace that that stack
+ is installed into.
+
+At install time:
+ - Cluster Stack: generate a set of `ClusterRole` objects that will be aggregated to the environment
+  admin, edit, view `ClusterRole` objects.
+ - Namespaced Stack: generate a set of `ClusterRole` objects that will be aggregated to the namespace
+  specific admin, edit, view `ClusterRole` objects.
+
+#### Default Crossplane ClusterRoles
+
+The following built in roles will be installed and packaged with Crossplane, and updated with role aggregation as we
+ install additional stacks.
+
+The verbs supported in these roles(admin,edit,view) will model behavior based on existing logic around
+ [built-in kubernetes roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings).
+
+|ClusterRole Name | Permissions |
+|:---------------------|:----------|
+| crossplane-admin | Admin: ClusterStacks, Crossplane CRDs, Namespaces, (Cluster)RoleBindings, (Cluster)Roles |
+| crossplane-env-admin | Admin: ClusterStacks, globally scoped Crossplane CRDs |
+| crossplane-env-edit | Edit: ClusterStacks, globally scoped Crossplane CRDs |
+| crossplane-env-view | View: Globally scoped Crossplane CRDs, Not permitted: secrets |
+
+We will provide a convenience group - crossplane:master - that binds to crossplane-admin with a
+ ClusterRoleBinding. This mirrors Kubernetes functionality for cluster-admin and system:master. At install time you will
+ then be able to impersonate a group that is bound to crossplane-admin.
+
+Example running a command as crossplane-admin
+```
+kubectl {any crossplane-admin operation} --as-group=crossplane:master
+```
+
+For discoverability of the roles for the environment we will add the following labels:
+ * `crossplane.io/scope: "environment"`
+
+Match labels on these roles for auto-aggregation:
+ * `rbac.crossplane.io/aggregate-to-crossplane-{install-scope}-{role}: "true"`
+
+#### ClusterRoles for a given app namespace.
+
+For any namespace that is annotated with `rbac.crossplane.io/managed-roles: true`, the stack-manager will install and manage
+ the following roles for that namespace that will be properly configured for ClusterRole aggregation as outlined below.
+ These roles should be bound to subjects to grant access to a particular namespace.
+
+ * `crossplane:ns:{ns-name}:admin`
+ * `crossplane:ns:{ns-name}:edit`
+ * `crossplane:ns:{ns-name}:view`
+
+For discoverability of the roles for a given namespace we will add the following labels:
+ * `crossplane.io/scope: "namespace"`
+ * `namespace.crossplane.io/{namespace-name}: true`
+
+Match Roles:
+ * `rbac.crossplane.io/aggregate-to-{install-scope}-{role}: "true"`
```

Comment:
same here, since this is talking about a namespaced role, this match role statement example could be filled in with the install-scope right here in this doc, right? it's not variable, is it?

should it just always be "namespace", i.e. the value of `crossplane.io/scope: "namespace"` just above?